File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/JobPostRequest.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

import dev.jefy.connectpro.shared.application.dtos.BudgetData;
import dev.jefy.connectpro.shared.domain.vo.JobType;
import dev.jefy.connectpro.shared.domain.vo.WorkMode;

/**
 * @author Jôph Yamba
 */
public record JobPostRequest (
        UUID portfolioId,
        String title,
        String description,
        UUID categoryId,
        List<String> tags,
        BudgetData budget,
        JobType jobType,
        WorkMode workMode,
        List<String> spokenLanguages,
        LocalDate deadline
) {}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/JobPostRequest.kt
package dev.jefy.connectpro.portfolio.application.dtos

import dev.jefy.connectpro.shared.domain.vo.JobType
import dev.jefy.connectpro.shared.domain.vo.WorkMode
import java.util.*

data class JobPostRequest(
    val portfolioId: UUID,
    val name: String,
    val description: String,
    val categoryId: UUID,
    val tags: Set<String>,
    val jobType: JobType?,
    val workMode: WorkMode?,
){
    init {
        require(name.isNotBlank()) { "title must not be blank" }
        require(description.isNotBlank()) { "description must not be blank" }
    }
}